services/supervisor/src/transactions.rs

```diff
@@ -211,26 +212,19 @@ impl SupervisorInterface for Supervisor {
             .validator_id(author)
             .ok_or(Error::UnknownAuthor)?;
 
-        let entry = schema
-            .pending_proposal()
-            .get()
-            .ok_or_else(|| Error::ConfigProposeNotRegistered)?;
-
         // Verifies that this config proposal is registered.
-        if entry.propose_hash != vote.propose_hash {
-            return Err(Error::ConfigProposeNotRegistered.into());
-        }
-
-        let config_propose = entry.config_propose;
+        let proposal = schema.pending_propose_hashes().values().find(|proposal|
```

Comment:
This approach seems highly inefficient to me. If one validator node will be hacked and ton of proposes will be sent, it may stop the whole network.